.github/actions/compare-lockfiles/compareLockfiles.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const core = require('@actions/core');
+const semverLt = require('semver/functions/lt');
+const { parseLockfile, getLockfilePackageVersions } = require('./helpers');
+
+const currentLockfile = parseLockfile(core.getInput('current-lockfile-path') || 'current');
+const proposedLockfile = parseLockfile(core.getInput('proposed-lockfile-path') || 'proposed');
+
+// Exit script if different package-lock.json schema versions.
+if (proposedLockfile.lockfileVersion !== currentLockfile.lockfileVersion) {
+  console.log('package-lock.json versions do not match');
+  process.exit(1);
+}
+
+const currentPackageVersions = getLockfilePackageVersions(currentLockfile);
+const proposedPackageVersions = getLockfilePackageVersions(proposedLockfile);
+
+const regressions = proposedPackageVersions.filter(function(proposed) {
+  const current = currentPackageVersions.find(current => current.key === proposed.key);
+  const hasRegression = semverLt(proposed.version, current.version); // proposed.version < current.version.
```

Comment:
Should we split or group `dev` vs `non-dev` dependencies and checks as we might have more that one regressions per PR? I fear that this will create a lot of false positives.

Reply:
For example a simple update create numerous changes to `node_modules/@babel` packages that change their versions using minor versions: `7.18.6` -> `7.19.6`. How could we manage those?

Reply:
@theodesp The action currently looks for a lower version number following [semver](https://semver.org/) conventions. Also note that this only considers the version number of the packages within the [root packages object](https://github.com/wpengine/faustjs/blob/canary/package-lock.json#L5-L13434).

This action is not aware of any security vulnerabilities within a package, but rather just catching semver regressions.